### ISSUE TITLE
Inconsistent name across sections and pages

### ISSUE THREAD

**Issue by fcoveram** [WordPress/wporg-developer#526]
Initially reported [here](https://github.com/WordPress/Learn/issues/2622#issuecomment-2219810417) in [Learn#2622](https://github.com/WordPress/Learn/issues/2622)

Developer Resources is called differently across sections and pages. Here is a table pointing out the places where this occurs.

| Menu item in global header | Hero section in Homepage | Section name in local nav
|:---|:---|:---|
| Developers | Developer Resources | WordPress Developer Resources |

I would like @thetinyl's eyes on this and see which approach suits better. I'm drawn to keeping the global header and changing the local navigation to "Developer Resources."

**Comment by thetinyl** [comment]
> I'm drawn to keeping the global header and changing the local navigation to "Developer Resources."

Yes, definitely agree. Since visitors are already on WP.org, it should be implicit that the Developer Resources content is about WordPress.
